src/API/IJobs.cs

```diff
@@ -37,14 +38,20 @@ public interface IJobs
         /// <param name="pipelineId">Pipeline ID to create a new job from.</param>
         /// <param name="jobName">Name of the job.</param>
         /// <param name="jobPriority">Priority of the job.</param>
-        Task<Job> Create(string pipelineId, string jobName, JobPriority jobPriority);
+        Task<Job> Create(string pipelineId, string jobName, JobPriority jobPriority, IDictionary<string,string> metadata);
```

Comment:
nit: `IReadOnlyDictionary<TKey, TValue>` would be more correct, but not required.

Reply:
Will fix in upcoming PR